crates/settings/src/network.rs

```diff
@@ -45,9 +47,81 @@ impl Network {
         }
     }
 
+    pub fn parse_networks_from_yaml(
+        document: Arc<RwLock<StrictYaml>>,
+    ) -> Result<HashMap<String, Network>, YamlError> {
+        let document_read = document.read().map_err(|_| YamlError::ReadLockError)?;
+
+        let networks_hash = require_hash(
+            &document_read,
+            Some("networks"),
+            Some("missing field: networks".to_string()),
+        )?;
+
+        let networks = networks_hash
+            .into_iter()
+            .map(|(key, value)| {
+                let key = key.as_str().unwrap_or_default().to_string();
+                let network = Network {
+                    document: document.clone(),
```

Comment:
if you clone the document it will still hold the reference back to the original right?